### PR TITLE
DATA-1813 Fix KeyError when fromDataManagement is not in extra

### PR DESCRIPTION
**Pull request by kaywux**
Bug fix from https://github.com/viamrobotics/viam-python-sdk/pull/416

### CODE REVIEW

File: src/viam/utils.py [hunk 1]
```diff
@@ -258,8 +258,8 @@ async def to_thread(func: Callable[_P, _R], *args: _P.args, **kwargs: _P.kwargs)
 
 
 def from_dm_from_extra(extra: Optional[Dict[str, Any]]) -> bool:
-    """Used in modular filter components to get the 'fromDataManagement' value from an extra map."""
+    """Used in modular filter components to return whether 'fromDataManagement' was set in the extra map."""
     if extra is None:
         return False
 
-    return bool(extra["fromDataManagement"])
+    return "fromDataManagement" in extra
```
Comment (cheukt) [inline src/viam/utils.py:265]: isn't it possible that "fromDataManagement" is in extra but not true?

Comment (kaywux) [reply src/viam/utils.py]: The only time we pass "fromDM" is when it's explicitly set to true - but changed this to return the value.

**Review by benjirewis — APPROVED** [review]

File: src/viam/utils.py [hunk 2]
```diff
@@ -259,7 +259,7 @@ async def to_thread(func: Callable[_P, _R], *args: _P.args, **kwargs: _P.kwargs)
 
 def from_dm_from_extra(extra: Optional[Dict[str, Any]]) -> bool:
     """Used in modular filter components to get the 'fromDataManagement' value from an extra map."""
-    if extra is None:
+    if extra is None or "fromDataManagement" not in extra:
         return False
 
     return bool(extra["fromDataManagement"])
```
Comment (cheukt) [inline src/viam/utils.py:265]:
```suggestion
    return bool(extra.get("fromDataManagement", False))
```
feels a little more natural to me, but either is fine

**Review by cheukt — APPROVED** [review]

File: src/viam/utils.py (hunk 1 above)
Comment (cheukt) [reply src/viam/utils.py]: that sounds good, returning value is a bit more robust so thanks for making the change!

**Pull request merged**